File: 백준/Silver/15663. N과 M （9）/N과 M （9）.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)

var (
	n, m    int
	nums    []int
	visited []bool
	result  []int
	s       = bufio.NewScanner(os.Stdin)
	w       = bufio.NewWriter(os.Stdout)
)

func main() {
	defer w.Flush()
	s.Split(bufio.ScanLines)

	hw := scanLines()
	n, m = hw[0], hw[1] // <- 여기 수정

	nums = scanLines()
	sort.Ints(nums)

	visited = make([]bool, n)
	result = make([]int, 0, m)

	bt(0)
}

func scanLines() []int {
	s.Scan()
	parts := strings.Fields(s.Text())
	res := make([]int, len(parts))
	for i, part := range parts {
		res[i], _ = strconv.Atoi(part)
	}
	return res
}

func bt(depth int) {
	if depth == m {
		for _, v := range result {
			fmt.Fprintf(w, "%d ", v)
		}
		fmt.Fprintln(w)
		return
	}

	used := make(map[int]bool)

	for i, v := range nums {
		if visited[i] || used[v] {
			continue
		}

		visited[i] = true
		used[v] = true
		result = append(result, v)

		bt(depth + 1)

		result = result[:len(result)-1]
		visited[i] = false
	}
}

File: 백준/Silver/15663. N과 M （9）/N과 M （9）.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split("\n")

const [n, m] = input[0].split(" ").map(Number);
const nums = input[1].split(" ").map(Number).sort((a, b) => a - b)

let result = [];
let visited = new Array(n);
let answer = ""


function bt(depth) {
  if (depth == m) {

    result.map((el) => answer += el + " ")
    answer += "\n"
    return
  }

  let used = new Set()

  for (let i = 0; i < n; i++) {
    if (visited[i] || used.has(nums[i])) continue;

    visited[i] = true;
    result.push(nums[i])
    used.add(nums[i]);

    bt(depth + 1)

    visited[i] = false;
    result.pop()
  }

}

bt(0)

console.log(answer)